src/main/java/de/tum/cit/aet/artemis/modeling/service/ModelingExerciseFeedbackService.java

```diff
@@ -0,0 +1,152 @@
+     * @param modelingExercise the modeling exercise object.
+     */
+    public void generateAutomaticNonGradedFeedback(StudentParticipation participation, ModelingExercise modelingExercise) {
+        log.debug("Using athena to generate (modeling exercise) feedback request: {}", modelingExercise.getId());
+
+        // athena takes over the control here
+        var submissionOptional = participationService.findExerciseParticipationWithLatestSubmissionAndResultElseThrow(participation.getId()).findLatestSubmission();
+
+        if (submissionOptional.isEmpty()) {
+            throw new BadRequestAlertException("No legal submissions found", "submission", "noSubmission");
+        }
+        var submission = submissionOptional.get();
+
+        Result automaticResult = new Result();
+        automaticResult.setAssessmentType(AssessmentType.AUTOMATIC_ATHENA);
+        automaticResult.setRated(true);
+        automaticResult.setScore(0.0);
+        automaticResult.setSuccessful(null);
+        automaticResult.setSubmission(submission);
+        automaticResult.setParticipation(participation);
+        try {
+            this.resultWebsocketService.broadcastNewResult((Participation) participation, automaticResult);
+
+            log.debug("Submission id: {}", submission.getId());
+
+            var athenaResponse = this.athenaFeedbackSuggestionsService.orElseThrow().getModelingFeedbackSuggestions(modelingExercise, (ModelingSubmission) submission, false);
+
+            List<Feedback> feedbacks = athenaResponse.stream().filter(individualFeedbackItem -> individualFeedbackItem.description() != null).map(individualFeedbackItem -> {
+                var feedback = new Feedback();
+                feedback.setText(individualFeedbackItem.title());
+                feedback.setDetailText(individualFeedbackItem.description());
+                feedback.setHasLongFeedbackText(false);
+                feedback.setType(FeedbackType.AUTOMATIC);
+                feedback.setCredits(individualFeedbackItem.credits());
+                return feedback;
+            }).toList();
+
+            double totalFeedbacksScore = 0.0;
+            for (Feedback feedback : feedbacks) {
+                totalFeedbacksScore += feedback.getCredits();
+            }
+            totalFeedbacksScore = totalFeedbacksScore / modelingExercise.getMaxPoints() * 100;
+            automaticResult.setSuccessful(true);
+            automaticResult.setCompletionDate(ZonedDateTime.now());
+
+            automaticResult.setScore(Math.clamp(totalFeedbacksScore, 0, 100));
+
+            automaticResult = this.resultRepository.save(automaticResult);
+            resultService.storeFeedbackInResult(automaticResult, feedbacks, true);
+            submissionService.saveNewResult(submission, automaticResult);
+            this.resultWebsocketService.broadcastNewResult((Participation) participation, automaticResult);
```

Comment:
I'm not sure we need the cast here as `StudentParticipation` extends `Participation`